src/number_theory/arithmetic_function.lean

```diff
@@ -231,5 +242,269 @@ instance [comm_ring α] : comm_ring (arithmetic_function α) :=
 { .. arithmetic_function.add_comm_group,
   .. arithmetic_function.comm_semiring }
 
+section pointwise_mul
+
+/-- This is the pointwise product of `arithmetic_function`s. -/
+def pointwise_mul [mul_zero_class α] (f g : arithmetic_function α) :
```

Comment:
How about abbreviating this?
```suggestion
def pmul [mul_zero_class α] (f g : arithmetic_function α) :
```
I imagine that we would use this as `f.pmul g`, etc...